examples/core-hello-world/src/main/scala/com/mesosphere/usi/helloworld/CoreHelloWorld.scala

```diff
@@ -0,0 +1,166 @@
+    * change. So why is the input parameter of that Flow not simply a [[SpecUpdated]] but a tuple of
+    * [[SpecsSnapshot]] and Source[[SpecUpdated]]? This is because the first message sent to the scheduler
+    * by the framework should be a snapshot of all PodSpecs known to the framework - a [[SpecsSnapshot]]. Internally
+    * this will trigger [task reconciliation](http://mesos.apache.org/documentation/latest/reconciliation/) by the
+    * scheduler.
+    *
+    * The same logic applies for the output parameter of the scheduler Flow - it's a tuple of [[StateSnapshot]] and a
+    * Source of [[StateEvent]], which means that the first state event received by the framework will be a
+    * snapshot of all reconciled tasks states, followed by a individual updates.
+    *
+    * For more about the scheduler flow see [[Scheduler]] class documentation.
+    *
+    */
+  val schedulerFlow
+    : Flow[(SpecsSnapshot, Source[SpecUpdated, Any]), (StateSnapshot, Source[StateEvent, Any]), NotUsed] =
+    Scheduler.fromClient(client)
+
+  // Lets construct the initial specs snapshot which will contain our hello-world PodSpec. For that we generate
+  // - a unique PodId
+  // - a RunSpec with minimal resource requirements and hello-world shell command
+  // - a snapshot containing our PodSpec
+  val podId = PodId(s"hello-world.${UUID.randomUUID()}")
+  val runSpec = RunSpec(
+    resourceRequirements =
+      List(
+        ScalarResourceRequirement(ResourceType.CPUS, 0.1),
+        ScalarResourceRequirement(ResourceType.MEM, 32)),
+    shellCommand = """echo "Hello, world" && sleep 3600"""
+  )
+  val podSpec = PodSpec(
+    id = podId,
+    goal = Goal.Running,
+    runSpec = runSpec
+  )
+
+  val specsSnapshot = SpecsSnapshot(
+    podSpecs = Seq(podSpec),
+    reservationSpecs = Seq.empty
+  )
+
+  val completed = Source
+    // A trick to make our stream run, even after the initial element (snapshot) is consumed. We use Source.maybe
+    // which emits a materialized promise which when completed with a Some, that value will be produced downstream,
+    // followed by completion. To avoid stream completion we never complete the promise but prepend the stream with
+    // our snapshot together with an empty source for subsequent SpecUpdates (which we're not going to send)
+    .maybe
+    .prepend(Source.single(specsSnapshot))
+    .map(snapshot => (snapshot, Source.empty))
+    // Here our initial snapshot is going to the scheduler flow
+    .via(schedulerFlow)
+    // We flatten the output of the scheduler flow which is a tuple of an initial snapshot and a source of all
+    // later updates, into one stream where the first element is the snapshot and all later elements are single
+    // state events. This makes the event handling a simple match-case
+    .flatMapConcat {
+      case (snapshot, updates) =>
+        updates.prepend(Source.single(snapshot))
```

Comment:
I don't really get this. Why isn't the scheduler flow prepending for us, @timcharper  and @alenkacz?